src/issue.ts

```diff
@@ -26,14 +26,22 @@ export interface Comment {
   timestamp: number;
 }
 
-export interface Author {
+// This generic types allow us to create a closed union type that allows us to create exclusive or type unions.
+// https://github.com/Microsoft/TypeScript/issues/14094
+type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
+type XOR<T, U> = (T | U) extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;
+
+export type Author = XOR<Omit<Identity, "identity">, Omit<Identity, "urn">>;
```

Comment:
This seems like a really complex way of creating a sum type. What are you trying to achieve?

Reply:
Yeah I went down a rabbit hole to create an exclusive union type, but adding a type guard with an `Identity` enum similar to how it's handled in the http-api seems nice.

Reply:
You can create a sum type by simply tagging it though, eg. `{ kind: "resolved", name: string } | { kind: "unresolved": peer: string }`

Reply:
This should make it "exclusive" since the tag is required to be different on both sides.